### PR TITLE
Initial commit of adr-38 split components infrastructure

### PR DESCRIPTION
**Pull request by mikebell**

### CODE REVIEW

File: architecture-decision-record/038-split-components-infrastructure.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+Date: 2024-03-04
+
+## Status
+
+🤔 Incomplete
+
+## Context
+
+As part of the move to Kubernetes 1.25 PSP was deprecated and replaced with Gatekeeper. This meant a large change in the way our cluster operates and the modules at a components level. While upgrading it became clear that there are a large number of components dependent on each other that are not required. This makes the component level difficult to work with. We also found modules that used non standard methods of declaring dependency because of the version of terraform used at the time. Removing dependencies at the components level will allow us to focus on future features such as [multi cluster](036-multi-cluster.md) and blue green clusters.
```
Comment (sj-williams) [inline architecture-decision-record/038-split-components-infrastructure.md:11]:
```suggestion
Following the upgrade to Kubernetes 1.25, PodSecurityPolicy was removed, and the team chose to use Gatekeeper's mutating capabilities to replace the functionality that was previously provided by our PSP configurations. This meant a large change in the way our cluster operates and the modules at a components level. While upgrading it became clear that there are a large number of components dependent on each other that are not required. This makes the component level difficult to work with. We also found modules that used non standard methods of declaring dependency because of the version of terraform used at the time. Removing dependencies at the components level will allow us to focus on future features such as [multi cluster](036-multi-cluster.md) and blue green clusters.
```

File: architecture-decision-record/038-split-components-infrastructure.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+- EKS - Everything required to bring up a basic cluster.
+- Core - Core components that are required for our users to be able to safely and securely deploy their application to.
+- Components - Optional components that provide additional functionality to a cluster
+
+> Components are not optional in `live`. When testing new components and features all components should be used to make sure there are no edge case issues.
+
+### Considerations
+
+- Developing this using test clusters will be easier than `live`, documentation and a focus on how we deploy this to `live` should be at the front of the work.
```
Comment (sj-williams) [inline architecture-decision-record/038-split-components-infrastructure.md:23]:
```suggestion
- Developing this using test clusters will be easier than `live`, documentation and a focus on how we deploy this to `live`, and safely migrate existing terraform state into its new s3 location should be at the front of the work.
```

**Review by sj-williams — APPROVED** [review]
